frontend/src/Connection.ts

```diff
@@ -170,7 +176,8 @@ export class Connection {
     this.socket.send(`no-more-finality:${chain}`);
   }
 
-  public handleMessages = (messages: FeedMessage.Message[]) => {
+  private handleMessages = (messages: FeedMessage.Message[]) => {
+    this.messageTimeout?.reset();
```

Comment:
Reset the timeout timer each time a message is seen (and make this fn private because nothing external seemed to need access to it)

Reply:
I just went "wait, this will error if timeout is `None`" before I realized this is TypeScript.